mule-extensions-api/src/main/java/org/mule/runtime/extension/api/declaration/type/ExtensionsObjectFieldHandler.java

```diff
@@ -355,6 +359,13 @@ private void setOptionalAndDefault(Field field, ObjectFieldTypeBuilder fieldBuil
       }
     });
 
+    sdkOptionalAnnotation.ifPresent(optional -> {
+      fieldBuilder.required(false);
+      if (getDefaultValue(optional) != null) {
+        fieldBuilder.with(new DefaultValueAnnotation(optional.defaultValue()));
```

Comment:
I don't know if this would actually fail if both Optional annotations are present.

Reply:
Changed